### PR TITLE
Dashboard page

### PR DESCRIPTION
**Pull request by loganhz**
image

### CODE REVIEW

**Comment by vincent99** [comment]
Looks good, do you want to work on the cluster dashboard page?

3 gauges for cpu, memory, and pods (not storage, we don't have data):
  - Read from /v3-machines (filtered by clusterId)
  - `requested` on each machine is how much of that resource is "used", `allocatable` is the total allowed.  So if requested.cpu = 500m and allocatable.cpu = 940m, show 53%.
  - Add up all the requested and allocatables for each machine and divide to get the average for the cluster.  Separately save the smallest and largest percentage of any individual machine to show the tick marks.
  - While you're looping through machines, keep track of how many there are and how many are in active state for the next section.

On the right side, read from `componentStatuses` on the cluster and show 4 lines:
  - etcd (all componentStatuses with an id starting with `etcd-` have a condition of type=Healthy and status=True)
  - Controller (the componentStatus with id=controller-manager has healthy/true)
  - Scheduler (id="scheduler" ...)
  - Nodes (all the nodes you saw in the graph part are active)

At the bottom, if any of the conditions are not healthy, show a banner saying which thing is bad.  For nodes, show a banner for each node what is not active.

**Comment by loganhz** [comment]
Sure. I will do it on Tuesday once I get a customer project done.

**Comment by loganhz** [comment]
@vincent99 Do you mean `/v3/clusters/${cluserId}/schemas/node` ? There is an `allocatable` field in `node` schema. But there is no `requested` field there. So I am using some random value for the gauges.

**Comment by vincent99** [comment]
/v3/machines entries should have `allocable` and `requested`:

image

**Comment by loganhz** [comment]
@vincent99 done.

File: app/authenticated/cluster/index/route.js [hunk 1]
```diff
@@ -1,4 +1,17 @@
 import Route from '@ember/routing/route';
+import { inject as service } from '@ember/service';
 
 export default Route.extend({
+  store: service(),
+  scope: service(),
+
+  model: function () {
+    return this.get('store').findAll('machine', { filter: { clusterId: this.get('scope.currentCluster.id') } }).then((nodes) => {
```
Comment (vincent99) [inline app/authenticated/cluster/index/route.js]: globalStore

File: lib/shared/addon/utils/component-badge.js [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+
+const HEALTHY_TEXT_COLOR = '#23A454';
+const UNHEALTHY_TEXT_COLOR = '#EF494A';
+
+export default function initGraph(options) {
+  const { el, r, width, height, margin, fontSize } = getConfig(options);
+  const svg = d3.select(el).append('svg').attr('width', width).attr('height', height);
+
+  const rectangle = svg.append("rect").attr("x", r).attr("y", margin).attr("width", width).attr("height", height - margin*2)
+    .style("fill", options.healthy ? HEALTHY_BACKGROUND_COLOR: UNHEALTHY_BACKGROUND_COLOR);
```
Comment (vincent99) [inline lib/shared/addon/utils/component-badge.js]: Can these all be defined as classes so they can be different colors per theme?

File: translations/en-us.yaml [hunk 3]
```diff
@@ -696,6 +696,21 @@ containerSubpod:
...
+  cpu: CPU
+  memory: MEMORY
+  pods: PODS
+  subtitle: "{used} of {total} used"
+  node: Node
+  etcd: EtcD
+  scheduler: Scheduler
+  controllerManager: Controller Manager
+  value: Value
+  noNodes: There is no nodes.
```
Comment (vincent99) [inline translations/en-us.yaml]: There are no nodes.

File: lib/shared/addon/components/cluster-dashboard/component.js [hunk 4]
```diff
@@ -0,0 +1,170 @@
...
+    return this.get('nodes').filterBy('state', 'inactive');
+  }),
+
+  unhealthyComponents: computed('nodes.@each.{name,state}', function () {
+    return this.get('componentStatuses')
+      .filter(s => !s.conditions.any(c => c.type === 'Healthy' && c.status === 'True'));
+  }),
+
+  updateComponentsStatus: function() {
+    this.getComponents();
```
Comment (vincent99) [inline lib/shared/addon/components/cluster-dashboard/component.js]: This doesn't exist?

File: lib/shared/addon/components/cluster-dashboard/component.js [hunk 5]
```diff
@@ -0,0 +1,170 @@
...
+
+  setDashboard() {
+    const cpuGauge = this.getCpuGauge();
+    const memoryGauge = this.getMemoryGauge();
+    const podsGauge = this.getPodsGauge();
+    this.set('gauges', [cpuGauge, memoryGauge, podsGauge]);
+  },
+
+  getCpuGauge() {
+    return this.getGauge('cpu', n => n.allocatable.cpu * 1000, n => n.requested.cpu.substr(0, n.requested.cpu.length - 1));
```
Comment (vincent99) [inline lib/shared/addon/components/cluster-dashboard/component.js]: This (and memory below) should look at the unit suffix (`Mi

**Comment by loganhz** [comment]
@vincent99 updated.

**Pull request merged**